benchadapt/python/README.md

```diff
@@ -1,3 +1,129 @@
 # Python {benchadapt}
 
-A small python package with utilities for running Conbench benchmarks.
+A small python package with utilities for getting benchmark results into a Conbench server.
+
+## Useful components in this package
+
+### `BenchmarkResult` and `BenchmarkRun` dataclasses
+
+The `BenchmarkResult` and `BenchmarkRun` dataclasses are designed to make it easy to
+populate JSON payloads to post to a Conbench server. Their structure corresponds to the
+corresponding POST endpoint; they each have a `.to_publishable_dict()` method that
+produces a dict to post.
+
+Regardless of how you are using them, the docstrings of these two objects will be useful
+as you try to assemble your results to get them in Conbench. All fields are documented,
+as are interactions between them and what you likely need to specify.
+
+The objects try to help you fill in your payloads correctly, including some defaults,
+like populating `machine_info` with metadata on the current machine. If you are running
+on a cluster instead, you will need to populate `cluster_info` yourself, and
+`machine_info` will remain empty.
+
+There is light validation, but [for now] the server is the ultimate validator; it is
+possible to make payloads that will be rejected.
+
+If you are managing everything yourself, to actually send the results to a Conbench
+server, use
+[benchclients.ConbenchClient](https://github.com/conbench/conbench/blob/main/benchclients/python/benchclients/conbench.py).
+As benchclients is a dependency of benchadapt, you should not need to install anything
+new, and it is nicely set up to handle auth and such for you.
```

Comment:
Could you explain more here about what "managing everything yourself" means in this case?

Reply:
Reworded. It feels a little weird putting low-level API details before the higher-level adapters API, but in practice adapters make way more sense with some context around what `BenchmarkResult` looks like even though it's part of the lower-level API. 

Open to ideas if there's a better way to structure this; I don't want most users to use `ConbenchClient`, but nor do I want someone to go roll their own if they do want to interact with the API manually, so mentioning it seems important